src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -992,13 +992,18 @@ public long toNative(DynamicObject string,
             if (convertProfile.profile(currentRope instanceof NativeRope)) {
                 nativeRope = (NativeRope) currentRope;
             } else {
-                nativeRope = new NativeRope(getContext().getNativePlatform().getMemoryManager(), bytesNode.execute(currentRope), currentRope.getEncoding(), currentRope.characterLength());
+                nativeRope = makeNativeRope(bytesNode, currentRope);
                 Layouts.STRING.setRope(string, nativeRope);
             }
 
             return nativeRope.getNativePointer().address();
         }
 
+        @TruffleBoundary
+        private NativeRope makeNativeRope(RopeNodes.BytesNode bytesNode, final Rope currentRope) {
+            return new NativeRope(getContext().getNativePlatform().getMemoryManager(), bytesNode.execute(currentRope), currentRope.getEncoding(), currentRope.characterLength());
```

Comment:
All this code seems fine for PE, I think NativeRope's constructor should be marked with `@TruffleBoundary` instead.